### PR TITLE
Added Login Page & Login Functionality

### PR DESCRIPTION
**Pull request by zenovy**
> GET/POST handlers for /login page
> Login page on GET
> Real login check on POST

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+from app.controllers import HomeController, LoginController, LoginPageController
+
+from flask import request
 
 @app.route('/')
 def home():
     return HomeController().handle()
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
```
Comment (norseboar) [inline app/routes.py]: You can have two functions for the two methods, which seems a bit cleaner than having the if statement

e.g.
```
@app.route('/login', methods=['GET'])
def login_page():

@app.route('/login', methods=['POST'])
def login():
```

File: app/controllers.py [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
+
 class HomeController():
     def handle(self):
         user = User.query.first()
         if user:
             return f"Hello, {user.email}!"
         else:
             return "No users. Sad!"
+
+from app.password_util import validate_password
```
[the remark this answers is not in the document]
Comment (zenovy) [reply app/controllers.py]: Right, was going to keep imports bunched together with their controllers, but then didn't follow through. Looking back on it, probably not such a good idea anyways. Removed

File: app/controllers.py [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+
+class LoginController():
+    def handle(self):
+        req_data = request.form
+        email = req_data['email']
+        password = req_data['password']
+
+        user = User.query.filter(User.email == email).first() # email is unique
```
[the remark this answers is not in the document]
Comment (zenovy) [reply app/controllers.py]: Done

File: app/routes.py (hunk 1 above)
Comment (zenovy) [reply app/routes.py]: Done

**Pull request merged**